csharp/examples/IoTDataPipelineExample/Program.cs

```diff
@@ -75,14 +75,23 @@ public static async Task Main(string[] args)
                                     + recordBatch.Column(j).Data.NullCount);
                             }
 
-                            if (recordBatch.Schema.HasMetadata && recordBatch.Schema.Metadata.TryGetValue("SubjectId", out string subjectId))
+                            var col = (Int32Array)recordBatch.Column(0);
+                            var subjectId = col.Values[0].ToString();
+
+                            if (!recordBatchDict.ContainsKey(subjectId))
                             {
-                                if (!recordBatchDict.ContainsKey(subjectId))
-                                {
-                                    recordBatchDict.Add(subjectId, new List<RecordBatch>());
-                                }
-                                recordBatchDict[subjectId].Add(recordBatch);
+                                recordBatchDict.Add(subjectId, new List<RecordBatch>());
                             }
+                            recordBatchDict[subjectId].Add(recordBatch);
+
+                            //if (recordBatch.Schema.HasMetadata && recordBatch.Schema.Metadata.TryGetValue("SubjectId", out string subjectId))
```

Comment:
@eerhardt,  

It seems that there is something weird with the schema metadata part: 

When building record batches, each record batch has an unique SubjectId as the schema metadata value, however, when writing the Arrow file and using the schema of the first record batch as the default arrow file schema, it seems all record batches' SubjectIds in their schema metadata are overridden by first record batch's SubjectIds.

I'm not sure what's going on here, could you review this?

Reply:
If you look at https://arrow.apache.org/docs/format/Columnar.html#ipc-streaming-format, you see that the SCHEMA is stored at the beginning of the file (in the "STREAMING FORMAT" section) outside of each RECORD BATCH. Thus every record batch in a file needs to have the same schema.

But your code is trying to add metadata to the schema that is specific for each record batch. In order to accomplish this, you would need a different file / arrow stream for every record batch that had different subject IDs.